x-pack/solutions/observability/plugins/metrics_data_access/server/lib/metrics/query.ts

```diff
@@ -37,6 +37,14 @@ export const query = async (
     },
   };
   const hasGroupBy = Array.isArray(options.groupBy) && options.groupBy.length > 0;
+  const groupInstanceFilter =
+    options.groupInstance?.reduce<Array<Record<string, unknown>>>((acc, group, index) => {
```

Comment:
I think this way it is easier to read, wdyt?
```suggestion
    options.groupInstance?.reduce<Record<string, unknown>[]>((acc, group, index) => {
```

Reply:
eslint auto corrects it back to `Array<Record<string, unknown>>`. I will refactor this function in another PR